zebra-network/src/address_book.rs

```diff
@@ -155,19 +185,22 @@ impl AddressBook {
         );
 
         if let Some(updated) = updated {
-            // If a node that we are directly connected to has changed to a client,
-            // remove it from the address book.
-            if updated.is_direct_client() && previous.is_some() {
```

Comment:
We can't remove peers that are recently live.

If we get that removed peer as a gossiped or alternate address, we'll reconnect to it within the liveness interval. (The proptests discovered this bug.)